### PR TITLE
Improve lsp3 IPFS options

### PR DESCRIPTION
**Pull request by CallumGrindle**
### What kind of change does this PR introduce (bug fix, feature, docs update, ...)?
improvement

### What is the current behaviour (you can also link to an open issue here)?
LSP3 data can be passed as a JSON object or a https:// json url

### What is the new behaviour (if this is a feature change)?
LSP3 data can now be passed as ipfs:// url or encoded json url

### CODE REVIEW

File: src/lib/classes/lsp3-universal-profile.ts [hunk 1]
```diff
@@ -48,9 +49,12 @@ export class LSP3UniversalProfile {
     contractDeploymentOptions?: ContractDeploymentOptions
   ) {
     // -1 > Run IPFS upload process in parallel with contract deployment
-    const lsp3Profile = profileDeploymentOptions.lsp3Profile
-      ? getLsp3ProfileDataUrl(profileDeploymentOptions.lsp3Profile)
-      : null;
+    let lsp3Profile: string | ProfileDataBeforeUpload | Promise<LSP3ProfileDataForEncoding> =
```
Comment (rryter) [inline src/lib/classes/lsp3-universal-profile.ts:52]: `Promise<LSP3ProfileDataForEncoding> 

where is this coming from?

**Review by rryter — APPROVED** [review]

**Comment by rryter** [comment]
Oh, Tests are missing 👍

File: src/lib/classes/lsp3-universal-profile.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (CallumGrindle) [reply src/lib/classes/lsp3-universal-profile.ts:52]: Ah indeed. Thanks I can change this :)

**Review by rryter — APPROVED** [review]

File: src/lib/classes/lsp3-universal-profile.spec.ts [hunk 2]
```diff
@@ -65,4 +65,32 @@ describe('LSP3UniversalProfile', () => {
       },
     });
   });
+  it.skip('should deploy and set LSP3Profile data', async () => {
```
Comment (Hugoo) [inline src/lib/classes/lsp3-universal-profile.spec.ts:68]: `.skip` is intended?

File: src/lib/classes/lsp3-universal-profile.spec.ts [hunk 3]
```diff
@@ -65,4 +65,32 @@ describe('LSP3UniversalProfile', () => {
...
+      {
+        libAddresses: {
+          erc725AccountInit: baseContracts.universalProfile.address,
+          universalReceiverDelegateInit: baseContracts.UniversalReceiverDelegate?.address,
+        },
+      }
+    );
+
+    const ownerAddress = await contracts.ERC725Account.owner();
+    expect(ownerAddress).toEqual('0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266');
```
Comment (Hugoo) [inline src/lib/classes/lsp3-universal-profile.spec.ts:88]: Duplicate value with line 76, you can put this in a variable and reuse same variable at line 76 and 88.
Will also allow to name these addresses which can be useful

File: src/lib/classes/lsp3-universal-profile.spec.ts [hunk 4]
```diff
@@ -65,4 +65,32 @@ describe('LSP3UniversalProfile', () => {
       },
     });
   });
+  it.skip('should deploy and set LSP3Profile data', async () => {
+    const myLSPFactory = new LSPFactory(
+      provider,
+      '0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80'
```
Comment (Hugoo) [inline src/lib/classes/lsp3-universal-profile.spec.ts:71]: Maybe you can put this in a variable too?

const PRIVATE_KEY....

File: src/lib/classes/lsp3-universal-profile.ts [hunk 5]
```diff
@@ -131,6 +129,7 @@ export class LSP3UniversalProfile {
       contractDeploymentOptions
     ).pipe(
       scan((accumulator: DeployedContracts, deploymentEvent: DeploymentEvent) => {
+        console.log(deploymentEvent);
```
Comment (Hugoo) [inline src/lib/classes/lsp3-universal-profile.ts:132]: is the console log intended?

File: src/lib/classes/lsp3-universal-profile.spec.ts (hunk 2 above)
Comment (CallumGrindle) [reply src/lib/classes/lsp3-universal-profile.spec.ts:68]: it is yes. This test is WIP as there was some problems with the Jest config causing type errors we couldn't get to the bottom of yet. Merged so we can continue with release

File: src/lib/classes/lsp3-universal-profile.ts (hunk 5 above)
Comment (CallumGrindle) [reply src/lib/classes/lsp3-universal-profile.ts:132]: nope :)

**Pull request merged**